utilities/pipelines/manageIssueForFailingPipelines/New-IssuesForFailingPipelines.ps1

```diff
@@ -0,0 +1,114 @@
+﻿<#
+.SYNOPSIS
+Check for failing pipelines and create issues for those, that are failing.
+
+.DESCRIPTION
+If a pipeline fails, a new issue will be created, with a link to the failed pipeline. If the issue is already existing, a comment will be added, if a new run failed (with the link for the new failed run). If a pipeline run succeeds and a issue is open for the failed run, it will be closed (and a link to the successful run is added to the issue).
+
+.PARAMETER repo
+Needs to have the structure "owner/repositioryName"
+
+.PARAMETER limitNumberOfRuns
+Optional. Number of recent runs to scan for failed runs. Default is 100.
+
+.PARAMETER limitInDays
+Optional. Only runs in the past selected days will be analyzed. Default is 2.
+
+.EXAMPLE
+New-IssuesForFailingPipelines -repo 'owner/repo01' -limitNumberOfRuns 100 -limitInDays 2 -Verbose -WhatIf
+
+.NOTES
+The function supports -WhatIf
+#>
+function New-IssuesForFailingPipelines {
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $repo,
+
+        [Parameter(Mandatory = $false)]
+        [int] $limitNumberOfRuns = 100,
+
+        [Parameter(Mandatory = $false)]
+        [int] $limitInDays = 2
+    )
+
+    $issues = gh issue list --state open --label 'automation,bug' --json 'title,url,body,comments' --repo $repo | ConvertFrom-Json -Depth 100
```

Comment:
Question, is the GitHub CLI automatically installed on runners? And would need a 'Begin' section to validate it exists and install it if not? So this module can be tested locally

Reply:
It appears to be installed by default - at least with MS hosted agents. But good point - we should probanly mention that in the description as a requirement, or implement a logic that validates the installation

Reply:
GH CLI is installed on the runners. If you run it locally you need to install it.

Reply:
I added a note

Reply:
Fine by me. Thoughts @ahmadabdalla ?

Reply:
I'm good with that too. Closing